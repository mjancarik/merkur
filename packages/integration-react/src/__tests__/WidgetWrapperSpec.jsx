import React from 'react';
import { shallow } from 'enzyme';

import {
  selectorToAttribute,
  SelectorIdentifierMap,
  WidgetWrapperComponent as WidgetWrapper,
} from '../WidgetWrapper';

describe('SelectorIdentifierMap', () => {
  it('should match snapshot', () => {
    expect(SelectorIdentifierMap).toMatchSnapshot();
  });
});

describe('selectorToAttribute()', () => {
  it.each([
    [null, {}],
    [undefined, {}],
    [0, {}],
    [false, {}],
    ['#container', { id: 'container' }],
    ['.container', { className: 'container' }],
    ['.e_dU.atm-list-ul.d_gX.container', { className: 'container' }],
    ['.e_dU#atm-list-ul...d_gX#d', { id: 'd' }],
    [
      '#main:first-child > div > div & pre .container',
      { className: 'container' },
    ],
  ])('should parse %j into %j', (input, expected) => {
    expect(selectorToAttribute(input)).toStrictEqual(expected);
  });
});

describe('WidgetWrapperComponent', () => {
  it('should render with SSR html primarily', () => {
    expect(
      shallow(
        <WidgetWrapper
          containerSelector=".container"
          html={'<div><h1>Hello world</h1></div>'}>
          <span>Fallback</span>
        </WidgetWrapper>
      )
    ).toMatchInlineSnapshot(`
      <div
        className="container"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<div><h1>Hello world</h1></div>",
          }
        }
      />
    `);
  });

  it('should render with children', () => {
    expect(
      shallow(
        <WidgetWrapper containerSelector="#container" html={''}>
          <span>Fallback</span>
        </WidgetWrapper>
      )
    ).toMatchInlineSnapshot(`
      <div
        id="container"
      >
        <span>
          Fallback
        </span>
      </div>
    `);
  });

  it('should process complex selector correctly', () => {
    expect(
      shallow(
        <WidgetWrapper containerSelector="#main:first-child > div > div & pre .container">
          <span>Fallback</span>
        </WidgetWrapper>
      )
    ).toMatchInlineSnapshot(`
      <div
        className="container"
      >
        <span>
          Fallback
        </span>
      </div>
    `);
  });
});
