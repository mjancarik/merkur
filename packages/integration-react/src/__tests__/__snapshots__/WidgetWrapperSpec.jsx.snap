// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectorIdentifierMap should match snapshot 1`] = `undefined`;
